Dockerfile

```diff
@@ -47,6 +47,43 @@ RUN \
 
 WORKDIR /CIRRUS-core
 
+# Python310 target
+FROM core_base AS python310
+ENV PYTHON_3_10_VERSION "3.10.14"
+RUN \
+        dnf groupinstall "Development Tools" -y && \
+        dnf install openssl-devel bzip2-devel libffi-devel sqlite-devel -y && \
+        cd /usr/local && \
+        wget https://www.python.org/ftp/python/${PYTHON_3_10_VERSION}/Python-${PYTHON_3_10_VERSION}.tgz && \
+        tar xzf Python-${PYTHON_3_10_VERSION}.tgz && cd Python-${PYTHON_3_10_VERSION} &&  \
+        ./configure --enable-optimizations \
+              --enable-shared \
+              --enable-loadable-sqlite-extensions \
+              --prefix /usr/local \
+              LDFLAGS=-Wl,-rpath=/usr/local/lib && \
+        make altinstall &&  \
+        update-alternatives --install /usr/bin/python3 python3 /usr/local/bin/python3.10 1 && \
+        python3 -m pip install boto3 setuptools
+
+# Python38 target
+FROM core_base AS python38
+ENV PYTHON_3_8_VERSION "3.8.16"
+RUN \
+        dnf groupinstall "Development Tools" -y && \
+        dnf install openssl-devel bzip2-devel libffi-devel sqlite-devel -y && \
```

Comment:
Is it possible to pull out common stuff like `dnf install openssl-devel bzip2-devel libffi-devel sqlite-devel -y && \`? I don't know the answer.

Reply:
Potentially. After what @mattp0 discovered + his recent changes, I think I am going to close this PR. This line in his changes `FROM public.ecr.aws/lambda/python:3.9 as python3` makes me think these changes would probably not work well. I am keeping my CIRRUS-DAAC PR open, though.